.github/workflows/python_package.yml

```diff
@@ -0,0 +1,89 @@
+name: Python-package
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+env:
+  CONDA_ENV: test-env
+  GITHUB_ACTIONS: 'true'
+
+jobs:
+  test:
+    name: ${{ matrix.task }} (${{ matrix.os }}, ${{ matrix.compiler }}, Python ${{ matrix.python_version }})
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: macOS-latest
+            task: regular
+            python_version: 3.7
+            compiler: gcc
```

Comment:
I think we don't need to specify compiler here. It will be enough to continue setting `COMPILER` variable depending on OS type.

Reply:
I see, ok

Reply:
done in https://github.com/microsoft/LightGBM/pull/3745/commits/1b6f357b33a6df1e0639c66f1952ea17e3350b01

Reply:
https://github.com/microsoft/LightGBM/pull/3745/commits/90ca7191363778471a5b2c426e8d3321e28cb7f5 sets the compiler for mac jobs to `clang`, let's see how it goes and then I'll switch them back.

build: https://github.com/microsoft/LightGBM/actions/runs/475893405

Reply:
it seems like all the Python jobs passed with `COMPILER=clang` (https://github.com/microsoft/LightGBM/actions/runs/475893405)

Checked the logs too just to be safe and I don't see anything that should have caused a build failure and didn't

Reply:
swapped compilers back in https://github.com/microsoft/LightGBM/pull/3745/commits/33cf19c16ad8de08e0a0022590c993ecb19dd11e

Reply:
ok yes this seemed like it worked ok: https://github.com/microsoft/LightGBM/actions/runs/475927918

Reply:
Very nice, indeed! Thank you for testing!